run_unittests.py

```diff
@@ -1194,6 +1195,17 @@ def test_sort_libpaths(self):
             ['/home/mesonuser/.local/lib/pkgconfig', '/usr/local/libdata/pkgconfig']),
             ['/home/mesonuser/.local/lib', '/usr/local/lib', '/usr/lib'])
 
+    def test_dependency_factory_order(self):
+        b = mesonbuild.dependencies.base
+        f = b.DependencyFactory(
+            'test_dep',
+            methods=[b.DependencyMethods.PKGCONFIG, b.DependencyMethods.CMAKE]
+        )
```

Comment:
Please also add a second `DependencyFactory` where the methods are reversed.

Reply:
Okay, I think I have the test correct now, and I'm not rebasing out merges.